src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```
@@ -193,64 +225,178 @@ We support
+            if (!string.IsNullOrEmpty(rawRequirementType) && !Enum.TryParse(rawRequirementType, ignoreCase: true, out requirementType))
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidConfigurationSetting,
+                    $"Invalid requirement type '{rawRequirementType}' for feature '{featureName}'.");
+            }
 
-            if (featureManagementConfigurationSection.Exists())
+            string rawEnabled = configurationSection[ConfigurationFields.EnabledKeyword];
+
+            bool enabled = false;
+
+            if (!string.IsNullOrEmpty(rawEnabled) && !bool.TryParse(rawEnabled, out enabled))
             {
-                return featureManagementConfigurationSection.GetChildren();
+                throw new FeatureManagementException(
+                        FeatureManagementError.InvalidConfigurationSetting,
+                        $"Invalid enabled '{rawEnabled}' for feature '{featureName}'.");
             }
 
-            //
-            // There is no "FeatureManagement" section in the configuration
-            if (RootConfigurationFallbackEnabled)
+            if (enabled)
+            {
+                IEnumerable<IConfigurationSection> filterSections = conditions.GetSection(ConfigurationFields.ClientFiltersSectionName).GetChildren();
+
+                if (filterSections.Any())
+                {
+                    foreach (IConfigurationSection section in filterSections)
+                    {
+                        //
+                        // Arrays in json such as "myKey": [ "some", "values" ]
+                        // Are accessed through the configuration system by using the array index as the property name, e.g. "myKey": { "0": "some", "1": "values" }
+                        if (int.TryParse(section.Key, out int _) && !string.IsNullOrEmpty(section[ConfigurationFields.LowercaseFeatureManagementSectionName]))
```

Should we throw an exception if the child of `client_filters` is not an array?

---

I think we should just ignore it since we cannot parse it.

---

Schema validation should be based on the schema
https://github.com/Azure/AppConfiguration/blob/main/docs/FeatureManagement/FeatureFlag.v1.1.0.schema.json